http.go

```diff
@@ -322,7 +322,87 @@ func httpPutViolation(w http.ResponseWriter, r *http.Request) {
 	// Force object field and type to match value specified in request path
 	v.Object = valstr
 	v.Type = typestr
-	httpPutViolationsInner(w, r, typestr, []ViolationRequest{v})
+	if err := putViolation(typestr, v); err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+	w.WriteHeader(http.StatusOK)
+	return
+}
+
+func putViolation(typestr string, v ViolationRequest) error {
+	v.Fixup(typestr)
+	// Force type field to match value specified in request path
+	v.Type = typestr
+	err := v.Validate()
+	if err != nil {
+		return err
+	}
+
+	if err = validateType(v.Type, v.Object); err != nil {
+		return err
+	}
+
+	rep, err := repGet(typestr, v.Object)
+	if err == redis.Nil {
+		rep = Reputation{
+			Object:     v.Object,
+			Type:       typestr,
+			Reputation: 100,
+		}
+	} else if err != nil {
+		return err
+	}
+
+	err = rep.Validate()
+	if err != nil {
+		return err
+	}
+
+	// If recovery suppression was specified add the correct timestamp to the reputation
+	// entry. Is suppression is already indicated, only update it if it results in a new
+	// timestamp that is beyond what the existing value is.
+	if v.SuppressRecovery > 0 {
+		nd := time.Now().UTC().Add(time.Second *
+			time.Duration(v.SuppressRecovery))
+		if rep.DecayAfter.IsZero() || rep.DecayAfter.Before(nd) {
+			rep.DecayAfter = nd
+		}
+	}
+
+	origRep := rep.Reputation
+	found, err := rep.applyViolation(v.Violation)
+	if err != nil {
+		if !found {
```

Comment:
I want to get rid of this and instead add the violation request to the list of failures, but I dont know if this is a contract breaking change with our customers @ameihm0912 @ajvb

Reply:
Yeah I think we should just leave this as is for now, the violation configuration in iprepd is not always deployed at the same time as the violation configuration in an application and the inverse is also true, so we should just accept unknown violations here and log them as such